src/LoggerFactory.ts

```diff
@@ -0,0 +1,101 @@
+import { Logger } from './logger';
+
+export enum Level {
+  Debug = 1,
+  Info = 2,
+  Warn = 3,
+  Error = 4,
+  Critical = 5
+}
+
+interface Slf {
+  _chain: Middleware[];
+  _queued: Event[];
+  _factory: Factory | null;
+  _logLevel: Level | null;
+  hasWarned: boolean;
+}
+
+const __slf: Slf = {
```

Comment:
This will create a new `__slf` each type the `LoggerFactory` is imported, whereas before it re-used the one set on the initial import. That might have negative impact.